server/src/main/java/org/opensearch/common/settings/SettingsModule.java

```diff
@@ -179,6 +179,29 @@ public void configure(Binder binder) {
         binder.bind(IndexScopedSettings.class).toInstance(indexScopedSettings);
     }
 
+    /**
+     * Dynamically registers a new Setting at Runtime. This method is mostly used by plugins/extensions
+     * to register new settings at runtime. Settings can be of Node Scope or Index Scope.
+     * @param setting which is being registered in the cluster.
+     * @return boolean value is set to true when successfully registered, else returns false
+     */
+    public boolean registerDynamicSetting(Setting<?> setting) {
+        try {
+            registerSetting(setting);
```

Comment:
This part we have not implemented in a reliable way: if `clusterSettings.registerSetting`  or `indexScopedSettings.registerSetting` fails later on, the `setting` will still be registered by `registerSetting` but not applied properly.